src/aws-codeartifact/repository.ts

```diff
@@ -0,0 +1,248 @@
+   */
+  readonly description?: string;
+  /**
+   * The connections to external repositories
+   *
+   * You can use the AWS CLI to connect your CodeArtifact repository to an external repository by adding an external connection directly to the repository. This will allow users connected to the CodeArtifact repository, or any of its downstream repositories, to fetch packages from the configured external repository. Each CodeArtifact repository can only have one external connection.
+   */
+  readonly externalConnection?: RepositoryConnection;
+}
+
+/**
+ * A new or imported Repository.
+ */
+abstract class RepositoryBase extends Resource implements IRepository {
+  public abstract repositoryArn: string;
+  public abstract repositoryName: string;
+  public abstract domain: IDomain;
+}
+
+export interface RepositoryAttributes {
+  readonly repositoryArn: string;
+  readonly repositoryName: string;
+  readonly domain: IDomain;
+}
+
+export class Repository extends RepositoryBase implements IRepository, ITaggableV2 {
+  public static fromRepositoryAttributes(scope: Construct, id: string, attrs: RepositoryAttributes): IRepository {
+    class Import extends RepositoryBase {
+      public readonly repositoryArn = attrs.repositoryArn;
+      public readonly repositoryName = attrs.repositoryName;
+      public readonly domain = attrs.domain;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public static fromRepositoryArn(scope: Construct, id: string, repositoryArn: string): IRepository {
+    const repositoryResourceArnParts = Stack.of(scope).splitArn(repositoryArn, ArnFormat.SLASH_RESOURCE_NAME);
+    if (
+      repositoryResourceArnParts.resource !== 'repository' ||
+      repositoryResourceArnParts.account === undefined ||
+      repositoryResourceArnParts.resourceName === undefined
+    ) {
+      throw new Error(`Expected a repository ARN, but got ${repositoryArn}`);
+    }
+    const repositoryNameParts = repositoryResourceArnParts.resourceName.split('/');
+    const domainName = repositoryNameParts[0];
+    const repositoryName = repositoryNameParts[1];
+
+    const domain = Domain.fromDomainArn(
+      scope,
+      'domain',
+      Stack.of(scope).formatArn({
+        resource: 'domain',
+        service: 'codeartifact',
+        resourceName: domainName,
+      }),
+    );
+
+    return Repository.fromRepositoryAttributes(scope, id, {
+      repositoryArn: repositoryArn,
+      repositoryName: repositoryName,
+      domain,
+    });
+  }
+
+  protected cfnResource: CfnRepository;
+  protected cfnResourceProps: CfnRepositoryProps;
+
+  readonly cdkTagManager: TagManager;
+  readonly repositoryArn: string;
+  readonly repositoryName: string;
+  readonly domain: IDomain;
+  /**
+   * Optional policy document that represents the resource policy of this repository
+   *
+   * If specified, addToResourcePolicy can be used to edit this policy.
+   * Otherwise this method will no-op.
+   */
+  protected readonly policy?: PolicyDocument;
```

Comment:
because this is always set in the constructor.

```suggestion
  protected readonly policy: PolicyDocument;
```

Reply:
Alternatively, we may not have to create this policy until the addToResourcePolicy method is called.